From c0496126023b56c84c554fe7046569aa909043e1 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 17 Nov 2008 10:24:39 +1000
Subject: [PATCH 40/41] EXA: avoid copy operations if no boxes in use

Simple fix for now, I'm sure damage shouldn't be calling us with nbox = 0.
(cherry picked from commit 8f8a9c19ad58768b07461a3f4bccea98f7c4f958)
---
 exa/exa_accel.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/exa/exa_accel.c b/exa/exa_accel.c
index 522224b..ccef744 100644
--- a/exa/exa_accel.c
+++ b/exa/exa_accel.c
@@ -401,6 +401,10 @@ exaCopyNtoN (DrawablePtr    pSrcDrawable,
     RegionPtr srcregion = NULL, dstregion = NULL;
     xRectangle *rects;
 
+    /* avoid doing copy operations if no boxes */
+    if (nbox == 0)
+	return;
+
     pSrcPixmap = exaGetDrawablePixmap (pSrcDrawable);
     pDstPixmap = exaGetDrawablePixmap (pDstDrawable);
 
-- 
1.6.0.4

